src/auto-evo/RunResults.cs

```diff
@@ -368,6 +328,155 @@ public long GetPopulationInPatch(Species species, Patch patch)
             }
         }
 
+        /// <summary>
+        ///   Computes the final population of all species, by patch.
+        /// </summary>
+        public Dictionary<Patch, Dictionary<Species, long>> GetPopulationsByPatch(bool resolveMigrations = false,
+            bool resolveSplits = false, bool includeNewSpecies = true)
+        {
+            var speciesInPatches = new Dictionary<Patch, Dictionary<Species, long>>();
+            foreach (var species in results.Keys)
+            {
+                var populations = GetPopulationsByPatch(species, resolveMigrations, resolveSplits, includeNewSpecies);
+                foreach (var patchEntry in populations)
+                {
+                    if (!speciesInPatches.ContainsKey(patchEntry.Key))
+                        speciesInPatches[patchEntry.Key] = new Dictionary<Species, long>();
+
+                    patchEntry.Value.ToList().ForEach(s =>
+                        speciesInPatches[patchEntry.Key].Add(s.Key, s.Value));
```

Comment:
That way this dictionary lookup also doesn't have to happen for each iteration of the ForEach.